services/src/handlers/layers.rs

```diff
@@ -455,3 +475,354 @@ async fn layer_handler<C: Context>(
 
     Ok(web::Json(collection))
 }
+
+#[derive(Debug, Deserialize, ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct AddLayerRequest {
+    pub layer: AddLayer,
+    #[schema(value_type = String, example = "05102bb3-a855-4a37-8a8a-30026a91fef1")]
+    pub collection_id: LayerCollectionId,
+}
+
+/// Add a new layer to a collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layerDb/layers",
+    request_body = AddLayerRequest,
+    responses(
+        (status = 200, description = "OK", body = IdResponse<LayerId>,
+            example = json!({
+                "layerId": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+            })
+        )
+    ),
+    security(
+        ("session_token" = [])
+    )
+)]
+async fn add_layer<C: Context>(
+    _session: AdminSession, // TODO: allow normal users to add layers to their stuff
+    ctx: web::Data<C>,
+    request: web::Json<AddLayerRequest>,
+) -> Result<web::Json<IdResponse<LayerId>>> {
+    let request = request.into_inner();
+
+    let collection_id = request.collection_id;
+    let add_layer = request.layer.validated()?;
+
+    let id = ctx
+        .layer_db_ref()
+        .add_layer(add_layer, &collection_id)
+        .await?;
+
+    Ok(web::Json(IdResponse { id }))
+}
+
+/// Add a new collection to an existing collection
+#[utoipa::path(
+    tag = "Layers",
+    post,
+    path = "/layerDb/collections/{collection}",
+    params(
+        ("collection" = LayerCollectionId, description = "Layer collection id", example = "05102bb3-a855-4a37-8a8a-30026a91fef1"),
+    ),
+    request_body = AddLayerCollection,
+    responses(
+        (status = 200, description = "OK", body = IdResponse<LayerCollectionId>,
+            example = json!({
+                "layerCollectionId": "36574dc3-560a-4b09-9d22-d5945f2b8093"
```

Comment:
the field name is now "id" only

Reply:
[d8f8191](https://github.com/geo-engine/geoengine/pull/676/commits/d8f8191357f2796124fddbfc00c8bcfb0a8e09db)